gensim/models/term_similarity.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 Vit Novotny <witiko@mail.muni.cz>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+"""
+This module provides classes that deal with term similarities.
+"""
+
+from itertools import chain
+import logging
+from math import sqrt
+
+import numpy as np
+from scipy import sparse
+
+from gensim.matutils import corpus2csc
+from gensim.utils import SaveLoad, is_corpus
+
+logger = logging.getLogger(__name__)
+
+
+class TermSimilarityIndex(SaveLoad):
```

Comment:
I don't think that `gensim.models` is the right place for it (same for `levenshtein`), we have `gensim.similarities` for this kind of stuff (exception only for `KeyedVectors`)

CC: @piskvorky

Reply:
The module placement is one of the things that I hoped we could discuss, since there is little documentation about where things should go. Both modules implemented term similarity models; if you think they would feel more at home in `gensim.similarities`, then to `gensim.similarities` they will go.

Reply:
Yes, I think better to place it in `gensim.similarities`, because
- This is purely about the similarity (and we already have submodule for it)
- This doesn't "quack" like a standard gensim model